### PR TITLE
Allows to set multiple base auth urls

### PR DESCRIPTION
**Pull request by itoys**
Allows to set multiple base auth urls

### CODE REVIEW

File: lib/chatops/controller.rb [hunk 1]
```diff
@@ -118,26 +118,30 @@ def ensure_user_given
...
+
+      body = request.raw_post || ""
+
+      valid = @chatops_urls.any? do |url|
+        signature_string = [url, @chatops_nonce, @chatops_timestamp, body].join("\n")
+        # We return this just to aid client debugging.
+        response.headers["Chatops-Signature-String"] = Base64.strict_encode64(signature_string)
+        if signature_valid?(Chatops.public_key, @chatops_signature, signature_string) ||
+            signature_valid?(Chatops.alt_public_key, @chatops_signature, signature_string)
+            return true
```
Comment (levenleven) [inline lib/chatops/controller.rb:131]: should this be `next true` or we could remove `return true if valid`?

**Review by levenleven — APPROVED** [review]
Looks good

> Ok, tests fail because a test app created with the rails, "~> 6", but the lib code works with any version, validated locally

Let's add `~> 6` requirement and bump gem's major. No reason to support older versions I think as they are EOL

File: lib/chatops/controller.rb (hunk 1 above)
Comment (itoys) [reply lib/chatops/controller.rb:131]: this is a block for `.any?` method, so `return true` looks valid here

**Comment by itoys** [comment]
> Let's add ~> 6 requirement and bump gem's major. No reason to support older versions I think as they are EOL

This will stick rails version to 6.\*, so the lib won't work with projects which use rails 7.\*

File: lib/chatops/controller.rb (hunk 1 above)
Comment (levenleven) [reply lib/chatops/controller.rb:131]: I think `return` will exit the function

Comment (levenleven) [reply lib/chatops/controller.rb:131]: image

image

Comment (itoys) [reply lib/chatops/controller.rb:131]: from the block, yes

Comment (levenleven) [reply lib/chatops/controller.rb:131]: so `valid` will be never true, that's my point

Comment (itoys) [reply lib/chatops/controller.rb]: Fixed ❤️

**Review by levenleven — APPROVED** [review]
LGTM

**Pull request merged**